lib/IRGen/GenBuiltin.cpp

```diff
@@ -1537,3 +1539,682 @@ void irgen::emitBuiltinCall(IRGenFunction &IGF, const BuiltinInfo &Builtin,
+                       getArraySize(arraySize, elementTI),
+                       getArraySpareBits(arraySize, elementTI),
+                       elementTI.getFixedAlignment(),
+                       elementTI.isTriviallyDestroyable(ResilienceExpansion::Maximal),
+                       elementTI.isCopyable(ResilienceExpansion::Maximal),
+                       elementTI.isFixedSize(ResilienceExpansion::Minimal),
+                       elementTI.isABIAccessible())
+  {
+  }
+  
+  unsigned getExplosionSize() const override {
+    return Element.getExplosionSize() * ArraySize;
+  }
+
+  void loadAsCopy(IRGenFunction &IGF, Address addr,
+                  Explosion &explosion) const override {
+    eachElementAddr(IGF, addr,
+                    [&](Address elementAddr) {
+                      Element.loadAsCopy(IGF, elementAddr, explosion);
+                    });
+  }
+
+  void loadAsTake(IRGenFunction &IGF, Address addr,
+                  Explosion &explosion) const override {
+    eachElementAddr(IGF, addr,
+                    [&](Address elementAddr) {
+                      Element.loadAsTake(IGF, elementAddr, explosion);
+                    });
+  }
+  
+  void assign(IRGenFunction &IGF, Explosion &explosion, Address addr,
+              bool isOutlined, SILType T) const override {
+    auto eltTy = getElementSILType(IGF.IGM, T);
+    eachElementAddr(IGF, addr,
+                    [&](Address elementAddr) {
+                      Element.assign(IGF, explosion, elementAddr, isOutlined,
+                                     eltTy);
+                    });
+  }
+
+  void initialize(IRGenFunction &IGF, Explosion &explosion, Address addr,
+                  bool isOutlined) const override {
+    eachElementAddr(IGF, addr,
+                    [&](Address elementAddr) {
+                      Element.initialize(IGF, explosion, elementAddr, isOutlined);
+                    });
+  }
+  
+  void reexplode(Explosion &sourceExplosion,
+                 Explosion &targetExplosion) const override {
+    eachElement([&]{
+      Element.reexplode(sourceExplosion, targetExplosion);
+    });
+  }
+  
+  void copy(IRGenFunction &IGF,
+            Explosion &sourceExplosion,
+            Explosion &targetExplosion,
+            Atomicity atomicity) const override {
+    eachElement([&]{
+      Element.copy(IGF, sourceExplosion, targetExplosion, atomicity);
+    });
+  }
+  
+  void consume(IRGenFunction &IGF, Explosion &explosion,
+               Atomicity atomicity,
+               SILType T) const override {
+    auto eltTy = getElementSILType(IGF.IGM, T);
+    eachElement([&]{
+      Element.consume(IGF, explosion, atomicity, eltTy);
+    });
+  }
+
+  void fixLifetime(IRGenFunction &IGF,
+                   Explosion &explosion) const override {
+    eachElement([&]{
+      Element.fixLifetime(IGF, explosion);
+    });
+  }
+
+  template<typename Body>
+  void eachElementOffset(Body &&body) const {
```

Comment:
Either comment the signature or use `llvm::function_ref`; also, this should definitely be a `uint64_t`.